autoload/neomake/makers/ft/sh.vim

```diff
@@ -4,14 +4,35 @@ function! neomake#makers#ft#sh#EnabledMakers() abort
     return ['sh', 'shellcheck']
 endfunction
 
-" IDEA: could be detected once from "shellcheck --help" (since older versions
-" support zsh, and newer might do so again).
-let s:shellcheck_supported = ['sh', 'bash', 'dash', 'ksh']
+function! s:shellcheck_determine_supported() abort
+    let s:shellcheck_supported = ['sh', 'bash', 'dash', 'ksh']
+
+    " zsh support was available from the first release of shellcheck and
+    " removed in version 0.3.6. Before shellcheck version 0.3.1 there was no
+    " way to get the version information.
+    let l:version_line = matchstr(split(system('shellcheck --version'), '\n'),
+                \ '^version:')
+    if v:shell_error != 0
+        let s:shellcheck_supported += ['zsh']
```

Comment:
Well I guess I can check for exit status 3 and check for `empty(l:version_line)`. That should support all current versions. I'll do that.